yamllint/config.py

```diff
@@ -112,11 +112,15 @@ def parse(self, raw_content):
             with fileinput.input(conf['ignore-from-file']) as f:
                 self.ignore = pathspec.PathSpec.from_lines('gitwildmatch', f)
         elif 'ignore' in conf:
-            if not isinstance(conf['ignore'], str):
+            if isinstance(conf['ignore'], str):
+                self.ignore = pathspec.PathSpec.from_lines(
+                    'gitwildmatch', conf['ignore'].splitlines())
+            elif isinstance(conf['ignore'], list):
+                self.ignore = pathspec.PathSpec.from_lines(
+                    'gitwildmatch', conf['ignore'])
```

Comment:
Yamllint would crash (instead of reporting `invalid config: ...`) if I provide it a list of integers. Can you add something like `... and all(isinstance(e, str) for e in conf['ignore'])`?

(same below)